test/test_proxy_tensor.py

```diff
@@ -110,6 +111,16 @@ def f_backward(x):
             assert inp.grad is None
             torch.testing.assert_close(traced_graph_out, f(inp))
 
+    def test_inplace_metadata(self):
+        def f(x):
+            x.unsqueeze_(-1)
```

Comment:
You most likely want to clone x before using it here. Otherwise every call changes the same Tensors inplace again and again?

Reply:
As richard pointed out, the problem is that it is very hard to understand what the text is doing as it modifies the shared input inplace.

Reply:
I already updated it in the recent commit.